dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/AssumeSolver.java

```diff
@@ -55,21 +55,27 @@ private void run() throws InterruptedException, SolverException, InvalidConfigur
         SymmetryEncoder symmetryEncoder = SymmetryEncoder.withContext(context);
 
         logger.info("Starting encoding using " + ctx.getVersion());
+        prover.write("; Program encoding");
         prover.addConstraint(programEncoder.encodeFullProgram());
+        prover.write("; Memory model encoding");
```

Comment:
Seeing that you only generate comments, a `prover.writeComment` method might be nicer to use (which automatically add the `; `-prefix). Also, I think writing anything but comments defeats the purpose of the class. But that is up to you.